control/subscription_group.go

```diff
@@ -353,23 +431,22 @@ func (s *subscriptionGroup) process(id string) (serrs []serror.SnapError) {
 		"metrics":    fmt.Sprintf("%+v", s.requestedMetrics),
 	}).Debug("gathered collectors")
 
+	// notice that requested plugins contains only processors and publishers
 	for _, plugin := range s.requestedPlugins {
-		//add processors and publishers to collectors just gathered
-		if plugin.TypeName() != core.CollectorPluginType.String() {
```

Comment:
To reviewers:
**To reviewers**:
This "if" statement is not needed because only processors and publishers are in [subscriptionGroup.requestedPlugin](https://github.com/intelsdi-x/snap/blob/master/control/subscription_group.go#L72). It does not change current behaviour. Only tests in [control/subscription_group_medium_test.go](https://github.com/intelsdi-x/snap/pull/1677/files#diff-f25602ea0928e1ac1bd0f400107ef53cL149) need to be updated to reflect real usage.